persist/sqldb/db_type.go

```diff
@@ -0,0 +1,81 @@
+package sqldb
+
+import (
+	"database/sql"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/go-sql-driver/mysql"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/selection"
+	"upper.io/db.v3"
+)
+
+type dbType string
+
+const (
+	MySQL    dbType = "mysql"
+	Postgres dbType = "postgres"
+)
+
+func dbTypeFor(session db.Database) dbType {
+	switch session.Driver().(*sql.DB).Driver().(type) {
+	case *mysql.MySQLDriver:
+		return MySQL
+	}
+	return Postgres
+}
+
+func (t dbType) labelsClause(requirements labels.Requirements) (db.Compound, error) {
+	var conds []db.Compound
+	for _, r := range requirements {
+		cond, err := t.requirementToCondition(r)
+		if err != nil {
+			return nil, err
+		}
+		conds = append(conds, cond)
+	}
+	return db.And(conds...), nil
+}
+
+func (t dbType) requirementToCondition(r labels.Requirement) (db.Compound, error) {
```

Comment:
It might be better move `LabelsClause` and `requirementToCondition ` to `workflow_archive.go`.

Reply:
Not sure about this - need to revisit the code.

Reply:
moved